sift/src/test/java/siftscience/android/SiftTest.java

```diff
@@ -581,6 +581,54 @@ public void testAppStateCollectorWithMockLocation() {
         assertEquals(mockLocationJson, event.androidAppState.location);
     }
 
+    @Test
+    public void testAppStateForceUpload() {
+        MemorySharedPreferences preferences = new MemorySharedPreferences();
+
+        Sift.Config config = new Sift.Config.Builder().withDisallowLocationCollection(true).build();
+
+        SiftImpl sift = new SiftImpl(mockContext(preferences), config, "",
+                false, mockTaskManager());
+        assertTrue(preferences.fields.isEmpty());
+
+        AppStateCollector appStateCollector = new AppStateCollector(sift, mockContext(preferences));
+        appStateCollector.collect();
+        // Collect twice because the first one gets uploaded and flushed
```

Comment:
For app state I don't believe that the double call to collect is required because the app state queue config is set to 8 events https://github.com/SiftScience/sift-android/blob/master/sift/src/main/java/siftscience/android/SiftImpl.java#L34

Reply:
I don't believe that you comment is accurate, ` // Collect twice because the first one gets uploaded and flushed` the first one is not uploaded, right? I would suggest only having one collect call.

Reply:
According to the queue upload criteria check method [https://github.com/SiftScience/sift-android/blob/master/sift/src/main/java/siftscience/android/Queue.java#L182](isReadyForUpload()) will return **true** for the first collected event, since the `lastUploadTimestamp` is 0 by default and queue is not empty regardless of what ever the value of `uploadWhenMoreThan`